### PR TITLE
Allow cluster domain override through jsonnet _config

### PR DESCRIPTION
**Pull request by zalegrala**
#### What this PR does

Here we add a config variable `cluster_domain` with a default value of `cluster.local` and use the variable everywhere.  This allows an easy integration with environments that don't use `cluster.local` as the top level domain.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/mimir/config.libsonnet [hunk 1]
```diff
@@ -2,6 +2,7 @@
   _config+: {
     namespace: error 'must define namespace',
     cluster: error 'must define cluster',
+    tld:: 'cluster.local',
```
Comment (pracucci) [inline operations/mimir/config.libsonnet:5]: WDYT if we call it `cluster_domain`?

**Review by pracucci — COMMENTED** [review]
I'm OK with this change. I left a comment about the variable name. Can you also fix the conflict and run `make build-jsonnet-tests`, please?

**Comment by zalegrala** [comment]
Thanks for the review @pracucci .  I'll rebase today.

**Comment by zalegrala** [comment]
It looks like there was some inconsistency with the trailing dot on the domain.  I updated to include the trailing dot in the domain and compiled the jsonnet.  Let me know if other changes are needed.

**Comment by pracucci** [comment]
@zalegrala I wanted to move forward, but I can't push a rebase myself to your branch (don't have privileges). Please do another rebase and `make build-jsonnet-tests` because of other changes we merged to `main` in the past few days. I'm going to enable the auto-merge, so that the PR will be merged rightaway once you've rebased.

Thanks!

**Review by pracucci — APPROVED** [review]

**Pull request merged**